Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -0,0 +1,438 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+#pragma once
+
+#include <optional>
+#include <variant>
+#include <SharedUtil.Template.h>
+#include "lua/CLuaOverloadParser.h"
+#include "lua/CLuaFunctionParseHelpers.h"
+#include "lua/CLuaStackChecker.h"
+#include "lua/LuaBasic.h"
+
+struct CLuaFunctionParserBase
+{
+    std::size_t iIndex = 1;
+    std::string strError = "";
+    std::string strErrorFoundType = "";
+
+    template <typename T>
```

Comment:
Not immediately obvious that this method is supposed to translate `std::variant<int, bool>` to `"int/bool"`. Could do with a comment here.

```suggestion
    // typeToNameVariant translates `std::variant<int, bool>` to `"int/bool"`
    template <typename T>
```